cluster-api/tf-deployer/CONTRIBUTING.md

```diff
@@ -0,0 +1,134 @@
+# Contributing Guidelines
+
+## Prerequisites
+
+### Install terraform
+
+Install terraform as per [instructions](https://www.terraform.io/intro/getting-started/install.html) or use the provided binary (`cluster-api/cloud/terraform/bin/terraform`).
+
+### Install Google Cloud SDK (gcloud)
+
+Google Cloud SDK (gcloud) will be helpful for two reasons:
+-  Inspect GCP resources during development;
+-  Set configuration values that will be used during development (like project name).
+
+Steps to follow:
+1.  Install as per [Cloud SDK instructions](https://cloud.google.com/sdk/);
+2.  Configure Cloud SDK to point to the GCP project you will be using.
+
+    ```bash
+    $ gcloud config set project <GCP_PROJECT_ID>
+    ```
+
+### Set GCP Credentials
+
+In order to develop the TF machine controller, you need to configure the credentials to push a newer version of the controller to GCR.
+
+You can set it in two ways, as explained below.
+
+#### Environment Variable GOOGLE_APPLICATION_CREDENTIALS
+
+Steps to follow:
+1. Verify that the environment variable `GOOGLE_APPLICATION_CREDENTIALS` is set pointing to valid service account credentials
+2. If not set, follow the [instructions on Google Cloud Platform site](https://cloud.google.com/docs/authentication/getting-started) to have it set up.
+
+#### Login Using Cloud SDK
+
+The alternative is to set the client credentials via gcloud by executing the command line below.
+
+```bash
+$ gcloud auth application-default login
+```
+
+### Install Docker
+
+1. Install [Docker](https://docs.docker.com/install/) on your machine;
+2. Make sure your user can execute docker commmands (without sudo). This is a way to test it:
+```bash
+$ docker run hello-world
+
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+...
+```
+
```

Comment:
Should probably mention `gcloud beta auth configure-docker` to make it work with gcr.